src/core/src/shade.rs

```diff
@@ -360,6 +360,8 @@ impl From<Stage> for Usage {
 pub struct AttributeVar {
     /// Name of this attribute.
     pub name: String,
+    /// Hack: Semantic index used by DirectX. Should be set to 0 in all other backends.
+    pub semantic_index: u32,
```

Comment:
this is a bit unfortunate - would require a new minor release for the whole project. Is there a way for us to only limit the changes to dx11 backend?

Reply:
ping @tangmi

Reply:
Hi! Sorry for the late reply!

The only thing I can think of is making `AttributeVar` parameterized over a "name" type that dx11 could specialize as `(String, u32)` and other backends would use `String`. This would only ensure that the size/layout of the data doesn't change for the other backends. I'm not sure if this is sufficient to avoid a minor release for whole project?

`DataLink::link_input` for `VertexBufferCommon` needs the semantic index, but it only gets `AttributeVar`. `Factory::create_pipeline_state_raw` also needs the semantic index, and `AttributeVar` seems like the only thing that makes sense to hold it.

Reply:
Sorry, I'm still trying to figure out what the best solution here is. So here is another question: would it be feasible for your case to require the user to *only* use semantic index == 0 in their shaders? It seems to me that it would be compatible with the current code. What would be the downside of this path?

Reply:
I'm not sure I understand the question fully, sorry!

Here's an alternate implementation that might be better: https://github.com/tangmi/gfx/pull/5 It doesn't change `AttributeVar` at the cost of very slightly higher cost marshaling in/out of that struct for the DX11 backend and slightly more complex vertex attribute handling on the DX11 backend.